components/kyma-environment-broker/internal/runtime/components_provider.go

```diff
@@ -0,0 +1,346 @@
+			err = fmt.Errorf("while trying to drain body reader: %w", err)
+		}
+		if closeErr := resp.Body.Close(); closeErr != nil {
+			err = fmt.Errorf("while trying to close response body reader: %w", err)
+		}
+	}()
+
+	if err = p.checkStatusCode(resp); err != nil {
+		return nil, err
+	}
+
+	type kymaComponents struct {
+		DefaultNamespace string          `yaml:"defaultNamespace"`
+		Prerequisites    []KymaComponent `yaml:"prerequisites"`
+		Components       []KymaComponent `yaml:"components"`
+	}
+
+	decoder := yaml.NewDecoder(resp.Body)
+
+	var kymaCmps kymaComponents
+	if err = decoder.Decode(&kymaCmps); err != nil {
+		return nil, fmt.Errorf("while decoding response body: %w", err)
+	}
+
+	requiredComponents := make([]KymaComponent, 0)
+	requiredComponents = append(requiredComponents, kymaCmps.Prerequisites...)
+	requiredComponents = append(requiredComponents, kymaCmps.Components...)
+
+	for i, cmp := range requiredComponents {
+		if cmp.Namespace == "" {
+			requiredComponents[i].Namespace = kymaCmps.DefaultNamespace
+		}
+	}
+
+	return requiredComponents, nil
+}
+
+func (p *defaultRequiredComponentsProvider) getComponentsYamlURL(kymaVersion string) string {
+	if p.isOnDemandRelease(kymaVersion) {
+		return fmt.Sprintf(onDemandComponentsURLFormat, kymaVersion)
+	}
+	return fmt.Sprintf(releaseComponentsURLFormat, kymaVersion)
+}
+
+// isOnDemandRelease returns true if the version is recognized as on-demand.
+//
+// Detection rules:
+//   For pull requests: PR-<number>
+//   For changes to the main branch: main-<commit_sha>
+//
+// source: https://github.com/kyma-project/test-infra/blob/main/docs/prow/prow-architecture.md#generate-development-artifacts
+func (p *defaultRequiredComponentsProvider) isOnDemandRelease(version string) bool {
+	isOnDemandVersion := strings.HasPrefix(version, "PR-") ||
+		strings.HasPrefix(version, "main-")
+	return isOnDemandVersion
+}
+
+func (p *defaultRequiredComponentsProvider) checkStatusCode(resp *http.Response) error {
+	if resp.StatusCode == http.StatusOK {
+		return nil
+	}
+
+	// limited buff to ready only ~4kb, so big response will not blowup our component
+	body, err := io.ReadAll(io.LimitReader(resp.Body, 4096))
+	if err != nil {
+		body = []byte(fmt.Sprintf("cannot read body, got error: %s", err))
+	}
+	msg := fmt.Sprintf("while checking response status code for Kyma components list: "+
+		"got unexpected status code, want %d, got %d, url: %s, body: %s",
+		http.StatusOK, resp.StatusCode, resp.Request.URL.String(), body)
+
+	switch {
+	case resp.StatusCode == http.StatusRequestTimeout:
+		return kebError.NewTemporaryError(msg)
+	case resp.StatusCode >= http.StatusInternalServerError:
+		return kebError.NewTemporaryError(msg)
+	default:
+		return errors.New(msg)
+	}
+}
+
+func (p *defaultAdditionalComponentsProvider) AdditionalComponents(kymaVersion internal.RuntimeVersionData, plan string) ([]KymaComponent, error) {
+	if kymaVersion.MajorVersion >= 2 {
```

Comment:
not necessary

Reply:
Let's remove that condition when we switch component providers. It doesn't interfere with the flow for Kyma 2 as it is now and in case someone tries provisioning with Kyma 1 the error will be thrown at the beginning of the process.